sale_order_line_input/models/sale_order.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2018 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    pricelist_id = fields.Many2one(
+        related="order_id.pricelist_id",
+        readonly=True,
+    )
+    force_company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Forced company",
+        compute="_compute_force_company_id",
+        readonly=False,
+        help="Technical field to force company or get it "
+        "from env user if order don't exist.",
+    )
+
+    @api.depends("order_id")
+    def _compute_force_company_id(self):
+        """Related company is not computed already when we click create new line"""
+        for line in self:
+            line.force_company_id = (
+                line.order_id.company_id
+                # Is not necessary use browse here
+                or self.env.context.get("force_company")
```

Comment:
force_company is deprecated. Moreover I'm not sure about the behavior

Reply:
@pedrobaeza

Reply:
`force_company` indeed it's deprecated. Now you look for the current company at `self.env.company`. What I don't know exactly is the goal of the field `force_company_id`. I suppose that is to transfer it to the `sale.order` header. @carlosdauden can you please clarify?

Reply:
Hi @carlosdauden did you check this? Thanks!

Reply:
The idea is that the user can enter orders from different companies in the same way as in the normal sales order view.

As the company_id field is a related of order_id.company_id and when starting to enter the order_id line does not exist, the force_company_id field allows the user to select the company so that the customers, products, taxes, etc. are those related to the selected company.

If force_company is never going to come in the context you could remove that line, but maybe some development is using this mechanism to inform the company.

Reply:
The idea is to enable the user to input orders from different companies in the same way they do from the regular sales order view.

Since the field company_id is a related field of order_id.company_id, and when starting to input the order_id line, it doesn't exist yet, the force_company_id field allows the user to select the company so that clients, products, taxes, etc., are associated with the chosen company.

If the force_company attribute is never going to be present in the context, you could consider removing that line. However, it's possible that some development might utilize this mechanism to specify the company.

Reply:
@carlosdauden thanks! @pedrobaeza should we keep this field?

Reply:
`force_company`, being deprecated now, shouldn't be kept, and you are already defaulting to `self.env.company`, so remove this line.

Reply:
@pedrobaeza Removed force_company_id